### PR TITLE
Implement attendees API on tickets/attendee

### LINKED ISSUE #644: Implement attendees API on tickets/attendees
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Implements attendees API on tickets/attendee

Fixes #644

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-751-open-event-frontend-fossasia.surge.sh

File: app/routes/events/view/tickets/attendees/list.js [hunk 1]
```diff
@@ -23,31 +23,108 @@ export default Route.extend({
...
-        'firstName'     : 'aajohn',
-        'lastName'      : 'bbDoe',
-        'ticketName'    : 'A'
-      }
-    ];
+    let filterOptions = [];
+    if (params.attendees_status === 'placed') {
+      filterOptions = [
+        {
+          name : 'status',
```
Comment (geekyd) [inline app/routes/events/view/tickets/attendees/list.js]: There is nothing called status in the API response. May be this is a typo.

File: app/routes/events/view/tickets/attendees/list.js [hunk 2]
```diff
@@ -23,31 +23,108 @@ export default Route.extend({
...
+              op   : 'eq',
+              val  : 'cancelled'
+            }
+          }
+        ];
+        break;
+      case 'checkedIn':
+        filterOptions = [
+          {
+            name : 'state',
```
Comment (geekyd) [inline app/routes/events/view/tickets/attendees/list.js]: Can we use `is-checked-in` here?

File: app/routes/events/view/tickets/attendees/list.js [hunk 3]
```diff
@@ -23,31 +23,108 @@ export default Route.extend({
...
-        'amount'        : 0,
-        'paidVia'       : 'Free',
-        'buyerName'     : 'sample@gmail.com',
-        'firstName'     : 'aajohn',
-        'lastName'      : 'bbDoe',
-        'ticketName'    : 'A'
-      }
-    ];
+    let filterOptions = [];
+    if (params.attendees_status === 'placed') {
```
Comment (geekyd) [inline app/routes/events/view/tickets/attendees/list.js]: We can minimise code duplication here
```
if(params.attendees_status === 'checkedIn' || params.attendees_status === 'notCheckedIn' || params.attendees_status === 'placed') {
  filterOptions = [
    {
      name : 'state',
      op   : 'eq',
      val  : params.attendees_status
    }
  ];
} else if (params.attendees_status === 'all') {
  filterOptions = [];
} else {
  filterOptions = [
    {
      name : 'order',
      op   : 'has',
      val  : {
        name : 'status',
        op   : 'eq',
        val  : params.attendees_status
      }
    }
  ];
}
```

File: app/routes/events/view/tickets/attendees/list.js [hunk 4]
```diff
@@ -23,31 +23,108 @@ export default Route.extend({
...
+      filterOptions = [
+        {
+          name : 'status',
+          op   : 'eq',
+          val  : params.orders_status
```
Comment (geekyd) [inline app/routes/events/view/tickets/attendees/list.js]: This should be `params.attendees_status`

**Comment by niranjan94** [comment]
Deployment link: https://pr-751-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@geekyd please merge this

**Comment by niranjan94** [comment]
Deployment link: https://pr-751-open-event-frontend-fossasia.surge.sh

**Pull request merged**